### PR TITLE
Added Rabin Karp Algorithm for Pattern Matching in Others folder

### PR DESCRIPTION
**Pull request by ReemaIsrani**

### CODE REVIEW

File: Others/RabinKarp.java [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+    /** generate a random 31 bit prime **/
+    private static long longRandomPrime() 
+    {
+        BigInteger prime = BigInteger.probablePrime(31, new Random());
+        return prime.longValue();
+    }
+    /** Main Function **/
+    public static void main(String[] args) throws IOException
+    {    
+        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
```
Comment (christianbender) [inline Others/RabinKarp.java]: Please use the ```Scanner``` class for this art of inputs. See [here](https://docs.oracle.com/javase/7/docs/api/java/util/Scanner.html)

File: Others/RabinKarp.java [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+        System.out.println("Rabin Karp Algorithm");
+        System.out.println("Enter Text\n");
+        String text = br.readLine();
+        System.out.println("Enter Pattern\n");
+        String pattern = br.readLine();
+        RabinKarp r = new RabinKarp();
```
Comment (christianbender) [inline Others/RabinKarp.java]: You don't need this lines. If you write the methods above as ```static```

File: Others/RabinKarp.java [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+ /** Large prime **/         
+    private long prime; 
+    /** radix **/         
+    private int radix;   
+    /** radix^(len-1) % prime **/        
+    private long rm;   	
+    /** pattern length **/
+    private int len;         
+ 
+    public void RabinKarpAlgo(String text, String pattern)
```
Comment (christianbender) [inline Others/RabinKarp.java]: In Java method names are in camel case. For instance ```rabinKarpAlgo```

File: Others/RabinKarp.java [hunk 4]
```diff
@@ -0,0 +1,101 @@
+
+import java.io.IOException;
```
Comment (christianbender) [inline Others/RabinKarp.java]: You import this header but don't used it.

**Comment by varunu28** [comment]
@ReemaIsrani Can you please do the changes requested? Otherwise you can close this PR and open a fresh one with changes implemented later

**Comment by prashantdoshi28** [comment]
Hi @ReemaIsrani , it seems this PR has not been updated since long time, so it will closed. If you want to contribute this code to repository, create a fresh PR. 

@ayaankhan98 I have added comment to notify author for creating fresh PR if needed, you can close it.

**Pull request closed without merging**